msdk-io/msdk-io-mzxml/src/main/java/io/github/msdk/io/mzxml/MzXMLFileParser.java

```diff
@@ -0,0 +1,318 @@
+                case TAG_PEAKS:
+                  double[] mzValues = new double[vars.peaksCount];
+                  float[] intensityValues = new float[vars.peaksCount];
+                  // Base64 decoder
+                  InputStream decodedIs = Base64.getDecoder().wrap(vars.peaksChars);
+                  InflaterInputStream iis = null;
+                  DataInputStream peakStream = null;
+
+                  // Decompress if the array is compressed
+                  if (vars.compressionFlag) {
+                    iis = new InflaterInputStream(decodedIs);
+                  }
+
+                  // Load the required input stream to peakStream
+                  if (iis == null)
+                    peakStream = new DataInputStream(decodedIs);
+                  else
+                    peakStream = new DataInputStream(iis);
+
+                  for (int i = 0; i < vars.peaksCount; i++) {
+
+                    // Always respect this order pairOrder="m/z-int"
+                    if ("64".equals(vars.precision)) {
+                      mzValues[i] = peakStream.readDouble();
+                      intensityValues[i] = (float) peakStream.readDouble();
+                    } else {
+                      mzValues[i] = (double) peakStream.readFloat();
+                      intensityValues[i] = peakStream.readFloat();
+                    }
+
+                  }
+
+                  // Set the final data points to the scan
+                  buildingScan.setDataPoints(mzValues, intensityValues, vars.peaksCount);
+
+                  // Auto-detect whether this scan is centroided
+                  MsSpectrumType spectrumType = SpectrumTypeDetectionAlgorithm
+                      .detectSpectrumType(mzValues, intensityValues, vars.peaksCount);
+                  buildingScan.setSpectrumType(spectrumType);
+                  break;
+              }
+              break;
+
+            case XMLStreamConstants.CHARACTERS:
+              if (vars.currentTag != null) {
+                switch (vars.currentTag.toString()) {
+                  case TAG_PEAKS:
+                    vars.peaksChars =
+                        new ByteArrayInputStream(xmlStreamReader.getText().toString().getBytes());
+                    break;
+                  case TAG_PRECURSOR_MZ:
+                    IsolationInfo newIsolation = new SimpleIsolationInfo(
+                        Range.singleton(xmlStreamReader.getText().toDouble()), null,
+                        xmlStreamReader.getText().toDouble(), vars.precursorCharge, null);
+                    buildingScan.getIsolations().add(newIsolation);
+                    break;
+                }
+              }
+              break;
+          }
+
+        } while (eventType != XMLStreamConstants.END_DOCUMENT);
+
+      } finally {
+        if (xmlStreamReader != null) {
+          xmlStreamReader.close();
+        }
+      }
+    } catch (IOException | XMLStreamException | javax.xml.stream.XMLStreamException
```

Comment:
In case like this you can just do 
````
} catch(Exception e) {
  throw (new MSDKException(e));
}
````
or even better:
````
} catch(Throwable e) {
  throw (new MSDKException(e));
}
````
That will also take care of runtime errors like OutOfMemoryError